utilities/mobile_server_pool.py

```diff
@@ -52,22 +66,33 @@ def get_nodes_from_pool_server(num_of_nodes, nodes_os_type, node_os_version, job
     :return: list of nodes reserved
     """
     # Getting list of available nodes through n1ql query
+    device_slave_ip = ""
+    if nodes_os_type in MOBILE_OS:
+        device_slave_ip = " AND slave_ip = '{}'".format(slave_ip)
     query_str = "select meta().id from `{}` where os='{}' " \
-                "AND os_version='{}' AND state='available'".format(BUCKET_NAME, nodes_os_type, node_os_version)
+                "AND os_version='{}' AND state='available'" \
+                " {}".format(BUCKET_NAME, nodes_os_type, node_os_version, device_slave_ip)
     query = cluster.query(query_str)
     pool_list = []
     for row in query:
         doc_id = row["id"]
-        print(doc_id, "doc id")
         is_node_reserved = reserve_node(doc_id, job_name_requesting_node)
-        vm_alive = check_vm_alive(doc_id)
-        if not vm_alive:
-            query_str = "update `{}` set state=\"VM_NOT_ALLIVE\" where meta().id='{}' " \
-                "and state='available'".format(doc_id)
-            query = cluster.query(query_str)
-        if is_node_reserved and vm_alive:
-            pool_list.append(str(doc_id))
-
+        if nodes_os_type not in MOBILE_OS:
+            vm_alive = check_vm_alive(doc_id)
+            if not vm_alive:
+                query_str = "update `{}` set state=\"VM_NOT_ALLIVE\" where meta().id='{}' " \
+                    "and state='available'".format(doc_id, doc_id)
+                query = cluster.query(query_str)
```

Comment:
Is query used anywhere? If not, maybe the "if not vm_alive" block could be removed?